### PR TITLE
commit for #951

### PR DESCRIPTION
**Pull request by ParkHanbum**
modify some call/jump instruction branch/indirect_branch flag.
added "XX_GRP_BR, XX_GRP_IND_BR" groups for branch/indirect_branch into X86.
added code to classify new groups.

after this commit accept, I will modify other architectures and binding.

1 | 2 | 3 | 4 | 5 | 6 | 7 | 8
-- | -- | -- | -- | -- | -- | -- | --
OPCODE | INSTRUCTION | Op/En | Desc | Table | instructionID(R) | instructionID(M) | instructionName
EB cb | JMP rel8 | D | jump short(?) |   |   |   |  
E9 cw | JMP rel16 | D | jump near, relative |   |   |   |  
E9 cd | JMP rel32 | D | jump near, relative |   |   |   |  
FF /4 | JMP r/m16 | M | jump near, absolute indirect | 1362 | 0x459 | 0x458 | JMP16r, JMP16m
FF /4 | JMP r/m32 | M | jump near, absolute indirect | 1001 | 0x45b | 0x45a | JMP32r, JMP32m
FF /4 | JMP r/m64 | M | jump near, absolute indirect | 1054 | 0x45d | 0x45c | JMP64r, JMP64m
EA cd | JMP ptr16:16 | D | jump far, absolute |   |   |   |  
EA cp | JMP ptr16:32 | D | jump far, absolute |   |   |   |  
FF /5 | JMP m16:16 | D | jump far, absolute indirect | 1362 | 0x2f5 |   |  
FF /5 | JMP m16:32 | D | jump far, absolute indirect | 1001 | 0x2f7 |   | FARJMP32m
REX.W + FF /5 | JMP m16:64 | D | jump far, absolute indirect | 1629 | 0x2f8 |   | FARJMP64
  |   |   |   |   |   |   |  
E8 cw | CALL rel16 | M | call near, relative |   |   |   |  
E8 cd | CALL rel32 | M | call near, relative |   |   |   |  
FF /2 | CALL r/m16 | M | call near, absolute indirect | 1362 | 0x162 | 0x161 | CALL16r, CALL16m
FF /2 | CALL r/m32 | M | call near, absolute indirect | 1001 | 0x163 | 0x164 | CALL32r, CALL32m
FF /2 | CALL r/m64 | M | call near, absolute indirect | 1054 | 0x167 | 0x165 | CALL64r, CALL64m
9A cd | CALL ptr16:16 | D | call far, absolute |   |   |   |  
9A cp | CALL ptr16:32 | D | call far, absolute |   |   |   |  
FF /3 | CALL m16:16 | M | call far, absolute indirect | 1362 | 0x2f0 |   |  
FF /3 | CALL m16:32 | M | call far, absolute indirect | 1001 | 0x2f2 |   | FARCALL32m
REX.W + FF /3 | CALL m16:64 | M | call far, absolute indirect | 1629 | 0x2f3 |   | FARCALL64

### CODE REVIEW

File: include/capstone/x86.h [hunk 1]
```diff
@@ -1899,6 +1899,10 @@ typedef enum  x86_insn_group {
 	X86_GRP_IRET,	// = CS_GRP_IRET
 	// all privileged instructions
 	X86_GRP_PRIVILEGE,	// = CS_GRP_PRIVILEGE
+	// relative and absolute branch groups
+	X86_GRP_BR,
+	// indirect branch groups
+	X86_GRP_IND_BR,
```
Comment (aquynh) [inline include/capstone/x86.h]: for clarity, i think we should just use longer names. what do you think about X86_GRP_BRANCH_INDIRECT & X86_GRP_BRANCH_DIRECT?

same change can be applied for CS_GRP_BR & CS_GRP_IND_BR.

Comment (ParkHanbum) [inline include/capstone/x86.h]: sounds reasonable. 
I will update other architecture's codes when update this.

**Comment by aquynh** [comment]
cool, what do you think about renaming INDIRECT_BRANCH to BRANCH_INDIRECT?

**Comment by aquynh** [comment]
ping ...

**Comment by ParkHanbum** [comment]
@aquynh 

alright, If you do not need it right away, I'll update you later on with other architectures.

**Comment by aquynh** [comment]
Sure, let me know when this is done, thanks.

**Comment by ParkHanbum** [comment]
@aquynh 

I'm modifying each architecture mapping file. 
on ARM64 and M64K , there is no logic to add GRP_JUMP to the get_insn_id function.
is that correct? not need this for this archtecture?

**Comment by aquynh** [comment]
you are right, we already have GRP_JUMP info in Mapping files, so dont need to do that in get_insn_id()

**Comment by ParkHanbum** [comment]
@aquynh  Can you review it? thanks

**Comment by aquynh** [comment]
this now has some conflicts with recent changes, can you take a look?

**Pull request closed without merging**